tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -71,36 +81,74 @@ object ScatterGather {
     /**
       * Produces a new [[Scatter]] from the existing scatter.  Partitions the the scattered objects of type Result
       * according to the given discriminator function.  The discriminator function maps an individual object of type
-      * Result to a key of type K.  The partitions are each a sequence of objects of type Result that share the same key
-      * of type K when the discriminator function is applied.
+      * Result to a key of type Key.  The partitions are each a sequence of objects of type Result that share the same key
+      * of type Key when the discriminator function is applied.
       *
       * The resulting Scatter[(Key, Seq[Result])] can be further mapped or gathered (or both!).
       */
     def groupBy[Key](f: Result => Key) : Scatter[(Key, Seq[Result])]
-  }
 
+    /**
+      * Converts this [[Scatter]] of scattered scatterers into a [[Scatter]] formed by the elements of the scatterers.
+      * In other words, this scatter must be consist of elements that themselves are scatters of the resulting element
+      * type.
+      *
+      * @param toScatter an implicit conversion which asserts that the element type this scatter is a [[Scatter]].
+      * @tparam NextResult the type of element of each scattered [[Scatter]]
+      */
+    def flatten[NextResult](implicit toScatter: Result => /*<:<!!!*/ Scatter[NextResult]): Scatter[NextResult]
+  }
 
   /**
    * Represents a gather operation in a scatter/gather pipeline.
    */
   sealed trait Gather[Source, Result <: Task] extends Task {
     private[ScatterGather] var sources: Option[Seq[Source]] = None
+    private[ScatterGather] var result: Option[Result] = None
+
+    /**
+      * Produces a new [[Scatter]] from the result of the [[Gather]].
+      * @param toPartitioner the partitioner that will partition [[Result]] into one or more [[NextResult]]s.
+      * @tparam NextResult the new type of the scatter
+      */
+    def scatter[NextResult](toPartitioner: Result => Partitioner[NextResult]): Scatter[NextResult] = {
```

Comment:
So when would you use this?  When you want to re-scatter but can't define the partitions until all the upstream operations up to the gather have actually executed?  I think this needs to be made really clear in the api doc that this defers creation of the partitioner and that's the only reason you should use it over just defining a new scatter and doing `gather ==> newScatter`.

Reply:
I find all the Gather.scatter() stuff rather complicated, and I'm yet to be convinced there is a non-strawman usecase for it.